src/util/sharing_map.h

```diff
@@ -535,6 +535,22 @@ class sharing_mapt
     return lp;
   }
 
+  /// Move a container node (containing a single leaf) further down the tree
+  /// such as to resolve a collision with another key-value pair. This method is
+  /// called by insert() to resolve a collision between a key-value pair to be
+  /// newly inserted, and a key-value pair existing in the map.
+  ///
+  /// \param i the depth of the inner node pointing to the container node with
```

Comment:
Coding guidelines recommend a colon after the parameter name. Speaking of which: `i` really isn't a good name, the only acceptable place for such a short name is a loop index IMHO.